resource/subfunctions/signed_program.md

```diff
@@ -0,0 +1,264 @@
+{: .fs-6 .fw-300 }
+
+&nbsp;
+
+{: .note }
+> _This may not be the only step to manage a **Signed Program** in your environment._
+> _You may also have to refresh the **PROGRAM** class to enact these changes._
+> _Please consult RACF documentation and manuals for an understanding of the **PROGRAM** class._
+
+## `ResourceAdmin.add_signed_program()`
+
+```python
+  def add_signed_program( self, signed_program_name: str, traits: dict = {} ) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+&nbsp;
+
+{: .experimental }
+> _Only a subset of available **Segments** and **Traits** are considered **Stable**. See [Segments](../../base/segments_traits_operators#segments) and [Traits](../../base/segments_traits_operators#traits) for more details._
+
+&nbsp;
+
+Define a new **Signed Program** profile in the **PROGRAM** class.
+
+#### 📥 Parameters
+* `signed_program_name`<br>
+  The name of the **Signed Program** profile being defined to the **PROGRAM** class.
+
+* `traits`<br>
+  A dictionary of **traits/attributes** that should be given to the resource on creation. See [Traits](../../base/segments_traits_operators#traits) to see what all of the valid **Resource Traits** are.
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result dictionary** or **Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **Return Code** of a **Security Result** returned by IRRSMO00 is **NOT** equal to `0`.
+* `AddOperationError`<br>
+  Raises `AddOperationError` when the **general resource profile** cannot be added because it already exists.
+* `SegmentTraitError`<br>
+  Raises `SegmentTraitError` when the dictionary of **traits/attributes** provided contains one or more **unknown** traits.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+from pyracf import ResourceAdmin
+resource_admin = ResourceAdmin()
+resource_admin.add_signed_program("TESTPRGM")
+{'securityResult': {'resource': {'name': 'TESTPRGM', 'class': 'PROGRAM', 'operation': 'set', 'requestId': 'ResourceRequest', 'commands': [{'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RDEFINE PROGRAM             (TESTPRGM) '}]}, 'returnCode': 0, 'reasonCode': 0}}
+```
+
+###### Security Result Dictionary as JSON
+```json
+{
+  "securityResult":{
+    "resource":{
+      "name":"TESTPRGM",
+      "class":"PROGRAM",
+      "operation":"set",
+      "requestId":"ResourceRequest",
+      "commands":[
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RDEFINE PROGRAM             (TESTPRGM) "
+        }
+      ]
+    },
+    "returnCode":0,
+    "reasonCode":0
+  }
+}
+```
+
+## `ResourceAdmin.alter_signed_program()`
+
+```python
+  def alter_signed_program( self, signed_program_name: str, traits: dict = {} ) -> Union[dict, bytes]:
+```

```

Comment:
```python
def alter_signed_program(self, signed_program_name: str, traits: dict = {}) -> Union[dict, bytes]:
```

Reply:
Leading spaces and unnecessary trailing spaces and indents removed across all function definitions.